doc/content/getting-started/migrating-from-v2/major-changes.md

```diff
@@ -0,0 +1,82 @@
+
+In order to connect a gateway to {{% tts %}}, follow instructions for [Adding a Gateway in the Console]({{< ref "/getting-started/console/create-gateway" >}}) or [Adding a Gateway using the CLI]({{< ref "/getting-started/cli#create-gateway" >}}). Then, reconfigure the gateway to connect to {{% tts %}}, and regenerate its API key (if required).
+
+Also see [Gateways]({{< ref "/gateways" >}}) for instructions on configuring popular LoRaWAN gateways with {{% tts %}}.
+
+## Application Data
+
+{{% tts %}} uses a different data format for uplink and downlink traffic than {{% ttnv2 %}}.
+
+For details on the data format of {{% ttnv2 %}}, see the documentation from [The Things Network](https://www.thethingsnetwork.org/docs/applications/mqtt/api.html).
+
+The data format has changed in {{% tts %}}. It uses a different schema, different names, and has much richer metadata support. Read more about that in [Data Formats]({{% ref "/integrations/data-formats" %}}).
+
+When migrating to {{% tts %}}, ensure your application can properly handle the new {{% tts %}} data format.
+
+### Payload Formats
+
+{{% ttnv2 %}} has support for payload decoders, converters, validators (for uplink) and encoders (for downlink). These can be either CayenneLPP or Javascript functions.
+
+{{% tts %}} has support for an uplink payload formatter (similar to the payload decoder) and a downlink payload formatter (similar to the payload encoder). These can be set per application, and can even be overridden per end device. Similar to {{% ttnv2 %}}, CayenneLPP and Javascript functions are supported.
+
+Migrating the {{% ttnv2 %}} payload encoder and decoder to an uplink and downlink payload formatter should be straightforward, since they have the same format.
+
+<!--
+TODO: https://github.com/TheThingsNetwork/lorawan-stack/issues/598
+Add a reference here once Payload Formatters are documented.
+-->
+
+### MQTT Traffic
+
+You will need to change the MQTT server your application connects to. {{% tts %}} has a new MQTT server address. You will also need to create API keys and update your MQTT credentials accordingly.
+
+[Read more about using the MQTT Server]({{< ref "/integrations/mqtt" >}}).
+
+### Integrations
+
+[Read about the Integrations supported by {{% tts %}}]({{< ref "/integrations" >}}).
+
+### Storage Integration
+
+{{% tts %}} does not currently support a Storage integration similar to {{% ttnv2 %}}. This feature will be added in a future release.
+
+## Suggested migration process
+
+First, update applications to support the {{% tts %}} data format. If you are using payload formatters, make sure to set them correctly from the Application settings page.
+
+Follow the rest of the guide and start by migrating a small number of test end devices (and gateways, if needed) to {{% tts %}}. Once you are confident that your devices are working properly, migrate the rest of your devices and gateways.
```

Comment:
Seeing a first, not seeing a second

Probably we want to advice people to setup a device on V3 first and test everything first, before migrating existing devices in the field